### DISCUSSION TITLE
Getting wrong version for textual[dev]

### DISCUSSION THREAD

**Discussion by amin0616** [Textualize/textual#2815] [Q&A] [answered]
Hey, I was having issues when trying to install the dev version of textual. When just downloading textual with 'pip install textual' it works great and downloads the most recent version 0.28.0. However, when I try installing the dev version of textual with 'pip install "textual[dev]" it downloads the 0.1.18 version and states: WARNING: textual 0.1.18 does not provide the extra 'dev'. and ends up downloading the 0.1.18 version. It also doesn't add a textual.exe at all. Thanks!

**Comment by davep** [comment]
Could you give us some more details about your environment? Things like operating system, Python version, how you've got Python installed, if you're using virtual environments, etc?

If possible, ideally a description of how we can create the same or a similar setup to what you're using and the exact commands used and the results seen, so we can test and compare.

**Comment by amin0616** [comment]
We installed Python 3.10.0 from python.org on a Windows 10 OS. Using VS Code, we created a virtual environment with py -m venv .venv. Within that environment, we pip install "textual[dev]". This pulls from our local PyPl collection.

**Comment by davep** [comment]
> This pulls from our local PyPl collection

So you're not pulling from PyPi itself, but some cache local to you?

What happens if you don't do that?

**Comment by amin0616** [comment]
Were working on an off network  computer so all of our pip installs come from a local pypi

**Comment by davep** [comment, marked as answer]
Given we've only got control over what is available *on* PyPi, I'm not sure what we can do here. The obvious question I'd be asking myself here when faced with your situation is *"is there something wrong with the content of our local PyPi?"* -- is this something you've looked into and eliminated?

**Comment by amin0616** [comment]
Our localpypi has both textual 0.1.18 and 0.28.0. We assumed it would download the most recent version. We also have a folder of textual_dev-0.0.2, is that up to date? Is there also a way to download a specific version with dev like how you can do: 

pip install "textual"==0.28.0

**Comment by TomJGooding** [comment]
Have you tried this?

```shell
pip install "textual[dev]==0.28.0"
```

**Comment by davep** [comment]
(as this seems to be a not-really-Textual issue, I'm converting this to a discussion).

**Comment by amin0616** [comment]
I get the error that it cannot find a version that satisfies the requirement msgpack>=1.0.3 so I updated it and believe it is now working, thanks for your time!

**Reply by TomJGooding** [reply]
Glad you managed to get this working. If you're installing packages from a local cache, then dependency versions are definitely something you'll need to watch for.
